pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MultiStageOperator.java

```diff
@@ -64,8 +65,15 @@ public TransferableBlock nextBlock() {
     try (InvocationScope ignored = Tracing.getTracer().createScope(getClass())) {
       _operatorStats.startTimer();
       TransferableBlock nextBlock = getNextBlock();
+
+      if (!_firstDataBlockEncountered && nextBlock.isNoOpBlock()) {
+        _operatorStats.resetStartTime();
+      } else {
+        _firstDataBlockEncountered = true;
+        _operatorStats.endTimer();
+      }
+
```

Comment:
Let's do it differently

```suggestion
      _operatorStats.startTimer();
      TransferableBlock nextBlock = getNextBlock();
      _operatorStats.endTimer(nextBlock);
```